app/src/main/java/com/android/calendar/widget/CalendarAppWidgetModel.java

```diff
@@ -80,13 +84,26 @@ public void buildFromCursor(Cursor cursor, String timeZone) {
             final boolean allDay = cursor.getInt(CalendarAppWidgetService.INDEX_ALL_DAY) != 0;
             long start = cursor.getLong(CalendarAppWidgetService.INDEX_BEGIN);
             long end = cursor.getLong(CalendarAppWidgetService.INDEX_END);
+            if (start == 0) {
```

Comment:
can you give some explanation here?

Reply:
some stuff as the begin = 0. task has no begin time so it will be recalculated